packages/playground/src/explorer/farms.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <view-layout>
+    <v-row>
+      <v-col>
+        <node-filters v-model="filterInputs" v-model:valid="isValidForm" @update:model-value="inputFiltersReset" />
+        <FarmsTable :items="farms" :loading="loading" v-model:selectedFarm="selectedFarm" @open-dialog="openDialog" />
+      </v-col>
+    </v-row>
+    <farmDialog v-if="selectedFarm" :openDialog="isDialogOpened" :farm="selectedFarm" @close-dialog="closeDialog" />
+  </view-layout>
+</template>
+
+<script lang="ts" setup>
+import type { CertificationType, FarmsQuery, PublicIp } from "@threefold/gridproxy_client";
+import { onMounted, ref, watch } from "vue";
+import { useRoute } from "vue-router";
+
+import router from "@/router";
+import { inputsInitializer } from "@/utils/filter_farms";
+
+import { getFarmQueries, getFarms } from "./utils/helpers";
+import {
+  type FarmFilterInputs,
+  type FarmFilterOptions,
+  type FarmMixedFilter,
+  farmOptionsInitializer,
+} from "./utils/types";
+
+export interface IFarm {
+  farmId: number;
+  name: string;
+  twinId: number;
+  pricingPolicyId: number;
+  certificationType: CertificationType;
+  publicIps: PublicIp[];
+  stellarAddress?: string;
+  totalPublicIp: number;
+  usedPublicIp: number;
+  freePublicIp: number;
+}
+const route = useRoute();
```

Comment:
The route here was created for what? I have a small suggestion, please watch the attached record.

[Screencast from 10-22-2023 06:05:55 PM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/aea49c2c-4156-4b50-83c8-b43ec2738ffc)

In the record, you push the `?twinId=<twin_id>` into the route after the second click after mount, also you push it in the first click on the row, I've tried several times and indeed it gets pushed in the first click only, which is unexpected behavior.

However, we really should push the `farmId` instead, and then after mounting the page we should check the rout if it includes the `farmId` we have to load the farm and open the dialog for the user, else we must close and clear any opened data.

Here's an example

```vue
<script lang="ts">
  const checkSelectedNode = async () => {
      if (route.query.farmId) {
        selectedFarmId.value = +route.query.farmId;
        isDialogOpened.value = true;
      }
    };
  
  onMounted(async () => {
      await checkSelectedNode();
    });
    
    watch(selectedFarmId, async () => await getFarmById())
</script>
```

This is an example, you must clear the route `onClose`.

```ts
  const closeDialog = () => {
    if (route.query.farmId) {
      router.replace(route.path);
    }
    isDialogOpened.value = false;
    selectedFarmId.value = undefined;
  };
```

So we will push the `farmId` instead and will automatically open the dialog `onMounted` if there is a `farmId` in the route.